core/web/log_controller.go

```diff
@@ -0,0 +1,46 @@
+package web
+
+import (
+	"net/http"
+
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+
+	"github.com/gin-gonic/gin"
+	"github.com/smartcontractkit/chainlink/core/services/chainlink"
+)
+
+// LogController manages the logger config
+type LogController struct {
+	App chainlink.Application
+}
+
+type loglevelPatchRequest struct {
+	EnableDebugLog *bool `json:"isDebugEnabled"`
+}
+
+type loglevelPatchResponse struct {
+	IsDebugEnabled bool `json:"isDebugEnabled"`
```

Comment:
I recently added a `presenters` package to web which contains the JSONAPI response objects for the API. Consider moving this response to the presenters package using something like this.

```
type LogResource struct {
	JAID
	DebugEnabled bool `json:"debugEnabled"`
}

// GetName implements the api2go EntityNamer interface
func (r LogResource) GetName() string {
	return "logs"
}
```

Note that the JSONAPI specification requires an `id`. The embedded `JAID` struct handles the id, and you could set a static id such as `log`.

```
LogResource{
	JAID: JAID{
		ID: "log"
	},
	DebugEnabled: *request.EnableDebugLog,
}
```

Reply:
Updated PR using above~